src/test/java/org/opentripplanner/test/support/VariableArgumentsProvider.java

```diff
@@ -50,6 +53,27 @@ private Stream<Arguments> getValue(Field field) {
 
     field.setAccessible(accessible);
 
-    return value == null ? null : (Stream<Arguments>) value;
+    if (value == null) {
+      return null;
+    } else if (value instanceof Collection<?> collection) {
+      return collection.stream().map(toArguments());
+    } else if (value instanceof Stream stream) {
+      return stream.map(toArguments());
+    } else {
+      throw new IllegalArgumentException(
+        "Cannot convert %s to stream.".formatted(value.getClass())
+      );
+    }
+  }
+
+  @Nonnull
+  private static Function<Object, Arguments> toArguments() {
+    return val -> {
+      if (val instanceof Arguments arguments) {
+        return arguments;
+      } else {
+        return Arguments.of(val);
+      }
+    };
```

Comment:
Is there a reason why we should not stick `@MethodSource` this instead?
```
  static List<TestCase> testCases() {}

  @ParameterizedTest
  @MethodSource("testCases")
  void test(TestCase testCase) ...

```
 - It allow for pair of test data and test to be together in the class, without having fields appaing between methods.
 - There is no framework method to maintain.
 - Related, `List`s are more robust than `Stream`s and avoid simple mistakes - can e.g. be reused in more than on test.
 - Using a class like `TestCase` or a list of arguments is both ok - whatever is best in each case.

Reply:
Note! static and none-static methods should also be grouped according to over code conventions, but I think we should change the conventions. A method can easily change between static and none static - it is also in some cases irrelevant from the callers point of view - at least if the caller is the instance itself.

Reply:
Actually, I had not noticed that with `@MethodSource` you don't need wrap your results into an `Arguments` instance, which is a major advantage so I agree that it has some advantages over our custom `VariableSource`.

I believe that I will start using `@MethodSource` from now on.

Reply:
Changed in 2ba658bdb78d9cd69fd967c0229d945f6d396889